### PR TITLE
set GOMAXPROCS=4 for the eks unit canary

### PR DESCRIPTION
**Pull request by tzneal**
Set GOMAXPROCS equal to the CPU limit.

### CODE REVIEW

File: config/jobs/kubernetes/sig-k8s-infra/periodics.yaml
Comment (sftim) [inline config/jobs/kubernetes/sig-k8s-infra/periodics.yaml:1]: LGTM

Maybe add a comment on the resource request / limits and on `env` that we should change 'em in lockstep.

**Review by xmudrii — APPROVED** [review]
This might fix https://github.com/kubernetes/kubernetes/issues/116990
/lgtm

**Comment by xmudrii** [comment]
@tzneal It failed because of yamllimt errors in the file that you changed. You need to expand the logs and search for the file.

**Pull request merged**